packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -316,8 +316,9 @@ export async function scanParentDirs(
       bunLockPath ? fs.readFile(bunLockPath, 'utf8') : null,
     ]);
 
-  // Priority order is bun with yarn lock > yarn > pnpm > npm > bun
-  if (bunLockBin && hasYarnLock) {
+  // Priority order is bun > yarn > pnpm > npm
+  // If no lockfile is detected, then npm is used.
+  if (bunLockBin) {
```

Comment:
<!-- Questions comments seek clarification.  -->
**question (non-blocking):** ‏should we change the special case precedence?

I think changing the precedence here of the special case (both a bun and yarn lockfile) is fine, but it's not strictly necessary for this PR. It only affects projects with multiple package manager lockfiles.

Why do you think we should change this precedence?

Reply:
Hmm, I think I misread this code (and I think the comment was confusing too). There are no early returns so the order is backwards (meaning last in the elseif takes precedence, not the first).

I'll just revert that part to avoid changing anything with the order but I do think there is a bug here we should revisit.